### ISSUE TITLE
fix: functions as properties should not be called during transformation 

### ISSUE THREAD

**Issue by FrancisS** [typestack/class-transformer#92]
I am trying to use this library with https://github.com/MikeMcl/decimal.js/ which at runtime has a property 'constructor' which is a function. TransformOperationExecutor.ts line 121 appears to blindly call any function it finds without any arguments which in this case causes an exception. 

I cannot easily decorate that property to be ignored. I have no use case for transforming function properties in this way and ultimately plan to use @Transform to convert the decimal to a number when running classToPlain. It would be nice if there was a way to globally disable this behavior.

**Comment by GeeWee** [comment]
I have the same issue. Apparently it only calls properties. 

E.g. if I define a function inside a class via:

```javascript
foo = () => {
  return 'foo'
}
```

it's called, but if I define it like

```javascript
foo(){
return 'foo'
```
it is not. 

(Note that I'm cheating a bit, as I'm actually using plainToClass to cast a class to another class of a different type - not sure if it matters.)

**Comment by ericmarcos** [comment]
That was causing a lot of problems to me too. The issue is the line 114 of TransformOperationExecutor.js:

```
else if (value[valueKey] instanceof Function) {
    subValue = value[valueKey](); --> Comment out this line
}
```

I don't know what's the point of calling all the functions but If I comment out this line everything works for me.

**Comment by ndce06** [comment]
+1

**Comment by NoNameProvided** [comment]
Fixed in 90feca36d2b44b6c66428ccef9402b13b58f7e2e.
